packages/aws-cdk-lib/aws-cloudfront-origins/lib/function-url-origin.ts

```diff
@@ -52,4 +78,84 @@ export class FunctionUrlOrigin extends cloudfront.OriginBase {
     };
   }
-}
\ No newline at end of file
+}
+
+/**
+ * An Origin for a Lambda Function URL with OAC.
+ */
+class FunctionUrlOriginWithOAC extends cloudfront.OriginBase {
+  private originAccessControl?: cloudfront.IOriginAccessControl;
+  private functionUrl: lambda.IFunctionUrl;
+  private readonly props: FunctionUrlOriginWithOACProps;
+
+  constructor(lambdaFunctionUrl: lambda.IFunctionUrl, props: FunctionUrlOriginWithOACProps = {}) {
+    const domainName = cdk.Fn.select(2, cdk.Fn.split('/', lambdaFunctionUrl.url));
+    super(domainName, props);
+    this.functionUrl = lambdaFunctionUrl;
+    this.originAccessControl = props?.originAccessControl;
+
+    this.props = props;
+
+    validateSecondsInRangeOrUndefined('readTimeout', 1, 180, props.readTimeout);
+    validateSecondsInRangeOrUndefined('keepaliveTimeout', 1, 180, props.keepaliveTimeout);
+  }
+
+  protected renderCustomOriginConfig(): cloudfront.CfnDistribution.CustomOriginConfigProperty | undefined {
+    return {
+      originSslProtocols: [cloudfront.OriginSslPolicy.TLS_V1_2],
+      originProtocolPolicy: cloudfront.OriginProtocolPolicy.HTTPS_ONLY,
+      originReadTimeout: this.props.readTimeout?.toSeconds(),
+      originKeepaliveTimeout: this.props.keepaliveTimeout?.toSeconds(),
+    };
+  }
+
+  public bind(scope: Construct, options: cloudfront.OriginBindOptions): cloudfront.OriginBindConfig {
+    const originBindConfig = super.bind(scope, options);
+
+    if (!this.originAccessControl) {
+      this.originAccessControl = new cloudfront.FunctionUrlOriginAccessControl(scope, 'FunctionUrlOriginAccessControl');
+    }
+
+    this.addInvokePermission(scope, options);
+
+    if (!this.isCheckAuthType()) {
+      cdk.Annotations.of(scope).addWarning(
+        'FunctionUrlOriginWithOAC: When the origin access control signing method is SIGV4_ALWAYS, it is recommended to set the authType of the Function URL to AWS_IAM.'
+      );
+    }
+
+    return {
+      ...originBindConfig,
+      originProperty: {
+        ...originBindConfig.originProperty!,
+        originAccessControlId: this.originAccessControl?.originAccessControlId,
+      },
+    };
+  }
+
+  private addInvokePermission(scope: Construct, options: cloudfront.OriginBindOptions) {
+    const distributionId = options.distributionId;
+
+    new lambda.CfnPermission(scope, `InvokeFromApiFor${options.originId}`, {
+      principal: 'cloudfront.amazonaws.com',
+      action: 'lambda:InvokeFunctionUrl',
+      functionName: this.functionUrl.functionArn,
+      sourceArn: `arn:${cdk.Aws.PARTITION}:cloudfront::${cdk.Aws.ACCOUNT_ID}:distribution/${distributionId}`,
+    });
+  }
+
+  /**
+   * Validation method: Ensures that when the OAC signing method is SIGV4_ALWAYS, the authType is set to AWS_IAM.
+   */
+  private isCheckAuthType(): boolean {
+
+    const cfnFunctionUrl = this.functionUrl.node.defaultChild as lambda.CfnUrl;
+    const CfnOriginAccessControl
```

Comment:
@gracelu0 
Sorry, I couldn't think of a concrete way to do it here, it can't be referenced from L2 Construct and the usual `this.something.node.defaultChild as ` doesn't reference this property!